python/tank/path_cache.py

```diff
@@ -730,35 +739,61 @@ def _get_filesystem_location_entities(self, folder_ids):
         """
 
         # get the ids that are missing from shotgun
-        # need to use this weird special filter syntax
+        batches = []
+
         if folder_ids:
```

Comment:
I think it should be
``` python
if folder_ids is not None:
```
Because the method _get_filesystem_location_entities has two behaviours:
- folder_ids=None -> collect all FileSystemLocations
- folder_ids=[...] -> collect FileSystemLocations for the given ids

If folder_ids is an empty list, it will collect all FileSystemLocations (for nothing).
And it's happening when you have to sync only Toolkit_Folders_Delete events, because in _do_incremental_sync, _get_filesystem_location_entities is called with the created_folder_ids list (what can be empty).

Reply:
thanks for pointing this out! @thebeeland, can you add a unit test to make sure we got coverage for that case?

Reply:
Done, and pushed to master. Bundling it for QA now.